test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemTest.scala

```diff
@@ -201,4 +201,79 @@ class SystemTest {
   @Test def getenv_should_link_and_does_not_throw(): Unit = {
     assertEquals(null, System.getenv(":${PATH}"))
   }
+
+  // System properties
+
+  private final val LineSeparatorPropName = "line.separator"
+  private final val ExistingPropName = "org.scalajs.testsuite.existingprop"
+  private final val TestPropName = "org.scalajs.testsuite.testprop"
+
+  @After def resetSystemPropertiesAfterEachTest(): Unit = {
+    System.setProperties(null)
+  }
+
+  @Test def getSetClearProperty(): Unit = {
```

Comment:
I find the two names here are a bit misleading. From what I understand is the first test tests the `dict` only case, the second test tests when forcing happens. Maybe add this in a comment to clarify?